responses/matchers.py

```diff
@@ -402,14 +403,27 @@ def header_matcher(
     :return: (func) matcher
     """
 
+    def __compare_with_regex(request_headers: Union[Dict[Any, Any], Any]) -> bool:
+        checked_request_headers = {k: 1 for k in request_headers.keys()}
+        for k, v in headers.items():
+            if request_headers.get(k) is not None:
```

Comment:
I assume this will crash if I add a header into my matcher that is not present in the request, or ?